internal/controller/notifier/notifier.go

```diff
@@ -47,6 +51,17 @@ func (watcher *Notifier) Register(ctx context.Context, worker string) (chan *rpc
 
 func (watcher *Notifier) Notify(ctx context.Context, worker string, msg *rpc.WatchInstruction) error {
 	slot, ok := watcher.workers.Load(worker)
+
+	deadline := time.Now().Add(watcher.workerWaitTimeout)
```

Comment:
What do you think about simply respecting the `ctx` here?

We already have a configurable timeout which defaults to 30 seconds and translates to a `context.Context`: https://github.com/cirruslabs/orchard/blob/88fba8004dd8b40c6abec5b61c6d595e8f246a5d/internal/controller/api_vms_ip.go#L29-L30

You'll just need to pass it to `Notify()` and respect it while waiting for the worker to re-connect.

Reply:
Yes, there are a couple of other `Notify()` invocations that need the proper context supplied to them, though.

Reply:
> With just waiting for context such use in the future might end up in a infinity loop.

Passing a time-bounded context should to the trick.

But overall it's a two sided coin: with the changes currently in this PR, we'll needlessly wait 30 seconds for each network-flipped worker in the scheduler, and there could be a lot of them.